src/components/CouponDetails/helpers.js

```diff
@@ -0,0 +1,64 @@
+/* eslint-disable import/prefer-default-export */
+import { SINGLE_USE, ONCE_PER_CUSTOMER } from '../../data/constants/coupons';
+import {
+  ACTION_LABELS, ACTION_TYPES, COUPON_FILTER_TYPES, FILTER_OPTIONS,
+} from './constants';
+
+export const getFilterOptions = (usageLimitation) => {
+  const shouldHidePartialRedeem = [SINGLE_USE, ONCE_PER_CUSTOMER].includes(usageLimitation);
+  let options = FILTER_OPTIONS;
+
+  if (shouldHidePartialRedeem) {
+    options = FILTER_OPTIONS.filter(option => option.value !== COUPON_FILTER_TYPES.partiallyRedeemed);
+  }
+
+  return options;
+};
+
+export const getFirstNonDisabledOption = (options) => {
+  const firstNonDisabledOption = options.find(option => !option.disabled);
+
+  if (firstNonDisabledOption) {
+    return firstNonDisabledOption.value;
+  }
+
+  return options[0].value;
+};
+
+export const getBASelectOptions = ({
+  isAssignView,
+  isRedeemedView,
+  hasTableData,
+  couponAvailable,
+  numUnassignedCodes,
+  numSelectedCodes,
+}) => ([{
+  label: ACTION_LABELS.assign,
+  value: ACTION_TYPES.assign,
+  disabled: !isAssignView || isRedeemedView || !hasTableData || !couponAvailable || numUnassignedCodes === 0, // eslint-disable-line max-len
+}, {
+  label: ACTION_LABELS.remind,
+  value: ACTION_TYPES.remind,
+  disabled: isAssignView || isRedeemedView || !hasTableData || !couponAvailable,
+}, {
+  label: ACTION_LABELS.revoke,
+  value: ACTION_TYPES.revoke,
+  disabled: isAssignView || isRedeemedView || !hasTableData || !couponAvailable || numSelectedCodes === 0, // eslint-disable-line max-len
+}]);
+
+export const shouldShowSelectAllStatusAlert = ({
+  tableData, hasAllCodesSelected, selectedToggle, selectedCodes,
+}) => {
+  if (!tableData || selectedToggle !== COUPON_FILTER_TYPES.unassigned) {
+    return false;
+  }
+
+  if (hasAllCodesSelected) {
+    return true;
+  }
+
+  return (
+    selectedCodes.length === tableData.results.length
+    && selectedCodes.length !== tableData.count
```

Comment:
I've read this 3 times and I still cannot make sense of it.
As we are returning based on these values being different, why wouldn't the tableData result set length equal the count of itself (tableData)?

Reply:
Honestly, I think in the case that `tableData.results.length` is different from `tableData.count` something has gone *seriously* wrong. I didn't change this code, just pulled it out and tested it. I'll to a `git blame` and see if the person who wrote it is still around to see if there's a good reason for it being this way...